app/models/declaration.rb

```diff
@@ -99,6 +101,22 @@ class Declaration < ApplicationRecord
   validates :ecf_id, uniqueness: { case_sensitive: false }
   validate :validate_max_statement_items_count
 
+  # TODO: When removing feature flag, set the optional: on the relationship to true instead
+  validates :delivery_partner, presence: true,
+                               if: -> { Feature.declarations_require_delivery_partner? }
+  validates :delivery_partner, inclusion: { in: :available_delivery_partners },
+                               if: -> { delivery_partner && delivery_partner_changed? }
+
+  validates :secondary_delivery_partner, absence: true, unless: :delivery_partner
+  validates :secondary_delivery_partner,
+            inclusion: { in: :available_delivery_partners },
+            if: -> { secondary_delivery_partner && delivery_partner_changed? }
```

Comment:
```suggestion
            if: -> { secondary_delivery_partner }
```
or if that second check is needed - it needs a spec.
should it be `secondary_delivery_partner_changed?`